scripts/main/contextMenu.js

```diff
@@ -403,41 +449,45 @@ contextMenu.photoTitle = function (albumID, photoID, e) {
 	basicContext.show(items, e.originalEvent, contextMenu.close);
 };
 
+/**
+ * @param {string} photoID
+ * @param {jQuery.Event} e
+ */
 contextMenu.photoMore = function (photoID, e) {
 	// Show download-item when
 	// a) We are allowed to upload to the album
 	// b) the photo is explicitly marked as downloadable (v4-only)
 	// c) or, the album is explicitly marked as downloadable
-	let showDownload =
-		album.isUploadable() ||
-		(photo.json.hasOwnProperty("downloadable")
-			? photo.json.downloadable === "1"
-			: album.json && album.json.downloadable && album.json.downloadable === "1");
-	let showFull = photo.json.url && photo.json.url !== "";
 
-	let items = [
+	// TODO: Check, if `hasOwnProperty` is really required. IMHO, `photo` always has this property.
```

Comment:
Yep, legacy code to handle Lychee v3 which didn't have it (see the comment just above). Can be removed.